src/state/slices/opportunitiesSlice/resolvers/cosmosSdk/utils.ts

```diff
@@ -100,3 +115,41 @@ export const makeAccountUserData = ({
     return acc
   }, {})
 }
+
+export const getDefaultValidatorAddressFromChainId = (chainId: ChainId) => {
+  switch (chainId) {
+    case cosmosChainId:
+      return SHAPESHIFT_COSMOS_VALIDATOR_ADDRESS
+    case osmosisChainId:
+      return SHAPESHIFT_OSMOSIS_VALIDATOR_ADDRESS
+    default:
+      throw new Error(`chainId ${chainId} is not a valid Cosmos SDK chainId`)
+  }
+}
+export const getDefaultValidatorAddressFromAssetId = flow([
+  (assetId: AssetId) => fromAssetId(assetId).chainId,
+  getDefaultValidatorAddressFromChainId,
+])
+
+export const getDefaultValidatorAddressFromAccountId = flow(
+  (accountId: AccountId) => fromAccountId(accountId).chainId,
+  getDefaultValidatorAddressFromChainId,
+)
+
+export const isCosmosUserStaking = (
+  userStakingOpportunity: UserStakingOpportunity,
+): userStakingOpportunity is CosmosSdkStakingSpecificUserStakingOpportunity =>
+  'undelegations' in userStakingOpportunity
+
+export const makeTotalUndelegations = (userStakingOpportunity: UserStakingOpportunity) =>
+  isCosmosUserStaking(userStakingOpportunity)
+    ? (userStakingOpportunity?.undelegations ?? []).reduce(
```

Comment:
as discussed can avoid guard and ternary if we just pass in `undelegations`?

Reply:
f8c17d792f001095d84a4b78d54c5271f29ec2e0